core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/deploysemantics/locationpreference/LocationPreference.scala

```diff
@@ -10,36 +9,8 @@ case class AddressInfo(
     controllerAddress: Address
 )
 
-trait LocationPreference extends Serializable {
+sealed trait LocationPreference extends Serializable
 
-  def getPreferredLocation(
-      addressInfo: AddressInfo,
-      physicalOp: PhysicalOp,
-      workerIndex: Int
-  ): Address
+object PreferController extends LocationPreference
 
-}
-
-class PreferController extends LocationPreference {
-  override def getPreferredLocation(
-      addressInfo: AddressInfo,
-      physicalOp: PhysicalOp,
-      workerIndex: Int
-  ): Address = {
-    addressInfo.controllerAddress
-  }
-}
-
-class RoundRobinPreference extends LocationPreference {
-  override def getPreferredLocation(
-      addressInfo: AddressInfo,
-      physicalOp: PhysicalOp,
-      workerIndex: Int
-  ): Address = {
-    assert(
-      addressInfo.allAddresses.nonEmpty,
-      "Execution failed to start, no available computation nodes"
-    )
-    addressInfo.allAddresses(workerIndex % addressInfo.allAddresses.length)
-  }
-}
+object RoundRobinPreference extends LocationPreference
```

Comment:
It is not very straight forward to me about the semantics. please add some comments and examples to explain this preference.

Reply:
done